src/ssl/PeekingPeerConnector.cc

```diff
@@ -196,6 +197,22 @@ Ssl::PeekingPeerConnector::initialize(Security::SessionPointer &serverSession)
             srvBio->setClientFeatures(details, cltBio->rBufData());
             srvBio->recordInput(true);
             srvBio->mode(csd->sslBumpMode);
+
+#if defined(SSL_OP_LEGACY_SERVER_CONNECT)
+            // While peeking, Squid is not generating any TLS bytes, but we are
+            // still being driven by OpenSSL negotiation logic. We enable as
+            // many features and workarounds as possible to reduce cases where
+            // OpenSSL refuses to accept a valid TLS server response. This code
+            // assumes that an admin should not expect a peeking Squid to
+            // automatically enforce a particular set of TLS conditions (e.g.,
+            // "no legacy TLS servers"). When that assumption is invalidated, we
+            // will need to add a configuration directive to set peeking TLS
+            // options.
+            if (csd->sslBumpMode == Ssl::bumpPeek) {
```

Comment:
Sooner or later, OpenSSL will stop supporting this option like it stopped supporting 

When that happens, the code below will do nothing but produce potentially misleading debugging -- some readers will assume that the option was applied when, in fact, it was not.

I was going to propose this change:

```suggestion
#if SSL_OP_LEGACY_SERVER_CONNECT
```

but then I noticed the following text on an OpenSSL manual page:

       The SSL_OP_ constants and the corresponding parameter and return values of the affected functions
       were changed to "uint64_t" type in OpenSSL 3.0.  For that reason it is no longer possible use the
       SSL_OP_ macro values in preprocessor "#if" conditions. However it is still possible to test whether
       these macros are defined or not.